discovery-infra/tests/test_cancel_and_reset.py

```diff
@@ -181,3 +187,197 @@ def test_cancel_reset_while_disable_workers(self, api_client, node_controller, c
+                cluster_id=cluster_id,
+                api_client=api_client
+            ), f'cluster {cluster_id} failed to reset from ' \
+               f'on attempt number: {i}'
+            self._reboot_hosts_and_wait_for_cluster_readiness(
+                api_client=api_client,
+                node_controller=node_controller,
+                cluster_id=cluster_id
+            )
+        self._install_ready_cluster(api_client, cluster_id)
+
+    @pytest.mark.regression
+    def test_reset_cluster_after_successful_installation(
+            self,
+            api_client,
+            node_controller,
+            cluster
+    ):
+        cluster_id = cluster().id
+        logger.debug(
+            'Cluster ID for '
+            'test_reset_cluster_while_at_least_one_node_finished_installation'
+            'is %s', cluster_id
+        )
+        self._prepare_cluster_for_installation(
+            api_client=api_client,
+            node_controller=node_controller,
+            cluster_id=cluster_id
+        )
+        self._install_ready_cluster(api_client, cluster_id)
+
+        with pytest.raises(Exception):
+            # TODO: catch the specific error
+            self.cancel_cluster_install(cluster_id, api_client)
+
+        with pytest.raises(Exception):
+            # TODO: catch the specific error
+            self.reset_cluster_install(cluster_id, api_client)
+
+    # TODO: Finish test
+    @pytest.mark.regression
+    def _test_reset_cluster_after_changing_cluster_configuration(
+            self,
+            api_client,
+            node_controller,
+            cluster
+    ):
+        cluster_id = cluster().id
+        logger.debug(
+            'Cluster ID for '
+            'test_reset_cluster_after_changing_cluster_configuration is %s',
+            cluster_id
+        )
+        self._prepare_cluster_for_installation(
+            api_client=api_client,
+            node_controller=node_controller,
+            cluster_id=cluster_id
+        )
+        self.start_cluster_install(cluster_id, api_client)
+        self.wait_for_one_host_to_be_installed(cluster_id, api_client)
+        self.cancel_cluster_install(cluster_id, api_client)
+        assert self.is_cluster_in_cancelled_status(
+            cluster_id=cluster_id,
+            api_client=api_client
+        ), f'cluster {cluster_id} failed to cancel'
+        self.reset_cluster_install(cluster_id, api_client)
+        assert self.is_cluster_in_insufficient_status(
+            cluster_id=cluster_id,
+            api_client=api_client
+        ), f'cluster {cluster_id} failed to reset from ' \
+           f'canceled state'
+        self._reboot_hosts_and_wait_for_cluster_readiness(
+            api_client=api_client,
+            node_controller=node_controller,
+            cluster_id=cluster_id
+        )
+
+        vips = node_controller.get_ingress_and_api_vips()
+        api_vip = IPNetwork(vips['api_vip'])
```

Comment:
Won't the installation fail if the DNS records aren't updated?

Reply:
Not sure, this test hasn't been completed and currently disabled by default